src/commands/auth-import.ts

```diff
@@ -0,0 +1,138 @@
+  .option("--block-size <blockSize>", "specify the block size (normally is 8) for standard scrypt.")
+  .option("--dk-len <dkLen>", "specify derived key length for standard scrypt.")
+  .option(
+    "--hash-input-order <hashInputOrder>",
+    "specify the order of password and salt. Possible values are SALT_FIRST and PASSWORD_FIRST. " +
+      "MD5, SHA1, SHA256, SHA512, HMAC_MD5, HMAC_SHA1, HMAC_SHA256, HMAC_SHA512 support this flag."
+  )
+  .before(requirePermissions, ["firebaseauth.users.create", "firebaseauth.users.update"])
+  .action(async (dataFile, options) => {
+    const projectId = needProjectId(options);
+    const checkRes = validateOptions(options);
+    if (!checkRes.valid) {
+      return checkRes;
+    }
+    const hashOptions = checkRes;
+
+    if (!_.endsWith(dataFile, ".csv") && !_.endsWith(dataFile, ".json")) {
+      return utils.reject("Data file must end with .csv or .json", { exit: 1 });
+    }
+    const stats = await fs.stat(dataFile);
+    const fileSizeInBytes = stats.size;
+    logger.info(`Processing ${clc.bold(dataFile)} (${fileSizeInBytes} bytes)`);
+
+    const batches: any[] = [];
+    let currentBatch: any[] = [];
+    let counter = 0;
+    let userListArr: any[] = [];
+    const inStream = fs.createReadStream(dataFile);
+    if (dataFile.endsWith(".csv")) {
+      userListArr = await new Promise<any[]>((resolve, reject) => {
+        const parser = parse();
+        parser
+          .on("readable", () => {
+            let record: string[] = [];
+            while ((record = parser.read()) !== null) {
+              counter++;
+              const trimmed = record.map((s) => {
+                const str = s.trim().replace(/^["|'](.*)["|']$/, "$1");
+                return str === "" ? undefined : str;
+              });
+              const user = transArrayToUser(trimmed);
+              const err = _.get(user, "error");
+              if (err) {
+                return reject(
+                  new FirebaseError(
+                    `Line ${counter} (${record.join(",")}) has invalid data format: ${err}`
+                  )
+                );
+              }
+              currentBatch.push(user);
+              if (currentBatch.length === MAX_BATCH_SIZE) {
+                batches.push(currentBatch);
+                currentBatch = [];
+              }
+            }
+          })
+          .on("end", () => {
+            if (currentBatch.length) {
+              batches.push(currentBatch);
+            }
+            resolve(batches);
+          });
+        inStream.pipe(parser);
+      });
+    } else {
+      userListArr = await new Promise<any[]>((resolve, reject) => {
+        const pipeline = new Chain([
+          Pick.withParser({ filter: /^users$/ }),
+          StreamArray.streamArray(),
+          ({ value }) => {
+            counter++;
+            const user = validateUserJson(value);
+            const err = _.get(user, "error");
+            if (err) {
```

Comment:
Same as above - prefer getting rid of previous line and making this `if (user.error)`